src/main/java/net/fabricmc/accesswidener/AccessWidenerReader.java

```diff
@@ -16,38 +16,39 @@
 
 package net.fabricmc.accesswidener;
 
-import java.io.BufferedReader;
 import java.io.IOException;
+import java.io.LineNumberReader;
+import java.io.Reader;
 import java.util.Locale;
 
 public final class AccessWidenerReader {
 	private final Visitor visitor;
 
-	public AccessWidenerReader(AccessWidener accessWidener) {
-		this.visitor = accessWidener;
-	}
-
 	public AccessWidenerReader(Visitor visitor) {
 		this.visitor = visitor;
 	}
 
-	public void read(BufferedReader reader) throws IOException {
+	public void read(Reader reader) throws IOException {
 		this.read(reader, null);
 	}
 
-	public void read(BufferedReader reader, String currentNamespace) throws IOException {
+	public void read(Reader reader, String currentNamespace) throws IOException {
```

Comment:
I don't think these descriptor changes are compatible at runtime. While this is fine at compile time, at runtime, the existing old user code will try to call `read` `(Ljava/io/BufferedReader;Ljava/lang/String;)V` and fail to find the method. Need a bridgegap measure to fix these.

Reply:
Yeah it's not, but I am not sure binary compatibility is actually a design goal here. But since it doesn't really cost anything to include the bridge method, why not 😁

Reply:
They should normally be version locked. BufferedReader is btw usually used explicitly when readLine is necessary.

Reply:
I wrap it in a LineNumberReader to get line number reporting, which is itself already a buffered reader, hence the change.
But since we're really only ever calling readLine(), we could probably get rid of it and just count the lines ourselves.

Reply:
Alright, I changed it back to take the BufferedReader and used internal line number counting instead of relying on the LineNumberReader. Also restored the old constructor.

Any of the more visible API changes we can do in a new version of AW that also supports V2 and limit this PR to internal changes / tests.